signer/keydbstore/cachedkeystore.go

```diff
@@ -0,0 +1,76 @@
+package keydbstore
+
+import (
+	"sync"
+
+	"github.com/docker/notary/trustmanager"
+	"github.com/docker/notary/tuf/data"
+)
+
+// Note:  once trustmanager's file KeyStore has been flattened, this can be moved to trustmanager
+
+type cachedKeyStore struct {
+	trustmanager.KeyStore
+	lock       *sync.Mutex
+	cachedKeys map[string]*cachedKey
+}
+
+type cachedKey struct {
+	role string
+	key  data.PrivateKey
+}
+
+// NewCachedKeyStore returns a new trustmanager.KeyStore that includes caching
+func NewCachedKeyStore(baseStore trustmanager.KeyStore) trustmanager.KeyStore {
```

Comment:
I'm thinking this could also replace a chunk of `trustmanager.GenericKeyStore` down the road, particularly if we get around to separating the password stuff out. We'd end up with something like `NewCacheKeyStore( NewGenericKeyStore( NewPasswdStorage( NewFileStore(keyDirectory) )))`   :-D